libr/arch/p/x86_nz/nzasm.c

```diff
@@ -5024,19 +5029,11 @@ static int parseOperand(RArchSession *a, const char *str, Operand *op, bool isre
 
 				// Reset nextpos: parseReg wants to parse from the beginning
 				nextpos = pos;
-				reg = parseReg (a, str, &nextpos, &reg_type, &op->rex_prefixed);
+				reg = parseReg (a, str, &nextpos, &reg_type, &op->extended, &op->rex_prefixed);
 
 				if (first_reg) {
-					op->extended = false;
-					if (reg > 8) {
-						op->extended = true;
-						op->reg = reg - 9;
-					} else {
-						op->reg = reg;
-					}
+					op->reg = reg;
 					first_reg = false;
-				} else if (reg > 8) {
-					op->reg = reg - 9;
```

Comment:
This fixes a mistake in `call [r10+rax*1]`, parsing `rax` will override saved `r10` in `op->reg`.